heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraCLIController.java

```diff
@@ -111,29 +113,46 @@ public void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove)
   }
 
   @Override
-  public void addContainers(Integer count) {
+  public Set<Integer> addContainers(Integer count) {
     //aurora job add <cluster>/<role>/<env>/<name>/<instance_id> <count>
     //clone instance 0
     List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "add", "--wait-until", "RUNNING", jobSpec + "/0", count.toString()));
-
-    if (isVerbose) {
-      auroraCmd.add("--verbose");
-    }
+        "aurora", "job", "add", "--wait-until", "RUNNING",
+        jobSpec + "/0", count.toString(), "--verbose"));
 
     LOG.info(String.format("Requesting %s new aurora containers %s", count, auroraCmd));
-    if (!runProcess(auroraCmd)) {
+    StringBuilder stderr = new StringBuilder();
+    if (!runProcess(auroraCmd, null, stderr)) {
       throw new RuntimeException("Failed to create " + count + " new aurora instances");
     }
+
+    if (stderr.length() <= 0) { // no container was added
+      throw new RuntimeException("empty output by Aurora");
+    }
+    return extractContainerIds(stderr.toString());
```

Comment:
the function signature is "String auroraStdoutStr" and here the parameter is stderr. Maybe the parameter should be renamed.

Reply:
update to `auroraOutputStr`